src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading;
+
+namespace System.Device.Pwm
+{
+    internal class UnixPwmChannel : PwmChannel
+    {
+        private readonly int _pwmChip;
+        private readonly int _pwmChannel;
+        private double _frequency;
+        private double _dutyCyclePercentage;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UnixPwmChannel"/> class.
+        /// </summary>
+        /// <param name="pwmChip">The PWM chip number.</param>
+        /// <param name="pwmChannel">The PWM channel number.</param>
+        /// <param name="frequency">The frequency in hertz.</param>
+        /// <param name="dutyCyclePercentage">The duty cycle percentage represented as a value between 0.0 and 1.0.</param>
+        /// <param name="polarity">The polarity the channel should start in.</param>
+        public UnixPwmChannel(
+            int pwmChip,
+            int pwmChannel,
+            double frequency,
+            double dutyCyclePercentage = 0.5,
+            PwmPolarity polarity = PwmPolarity.ActiveHigh)
+        {
+            _pwmChip = pwmChip;
+            _pwmChannel = pwmChannel;
+            ValidatePwmChannel();
+            OpenChannel();
+            Thread.Sleep(100);  // TODO: Need a better solution to delay for available file.
```

Comment:
I noticed the PWMx file isn't created fast enough.  I added this delay, but there needs to be a better way to check when file is available.  Thoughts?

Reply:
Try calling https://linux.die.net/man/2/sync instead of delay and see if you can see any difference. Assuming it work `sync` might be generally to slow, so you might want to expand File.WriteAllText into FileStream+StreamWriter and call `sysfs` (or perhaps even just explicit Flush will work).

Reply:
Generally to explain my theory:
- sync normally forces buffer cache to be written into disk - I suspect it might also work on virtual fs - that would force driver to process the virtual fs operation